src/met-atlas-viewer.js

```diff
@@ -1049,8 +1053,34 @@ function MetAtlasViewer(targetElement) {
     scene.background = new Color(color);
   }
 
+  /**
+   * Disposes the viewer
+   */
+  function dispose() {
+    // stop animation
+    cancelAnimationFrame(animationId);
+
+    // remove event listeners
+    renderer.domElement.addEventListener('dblclick', null, false);
+    window.removeEventListener('resize', onWindowResize, false);
+    window.removeEventListener('mousemove', onMouseMove, false);
+    window.removeEventListener('pointerdown', onMouseClick, false);
+    window.removeEventListener('keypress', onKeypress, false);
```

Comment:
I'm not getting the full picture here, so I hope it's okay to ask - how are these control-related listeners different from the ones below ? https://github.com/MetabolicAtlas/3d-network-viewer/blob/8e392604a6f161e1b819e8d7d2718135bea2b855/src/atlas-viewer-controls.js#L650-L666

Reply:
The four lines you quoted correspond to the following. https://github.com/MetabolicAtlas/3d-network-viewer/blob/8e392604a6f161e1b819e8d7d2718135bea2b855/src/met-atlas-viewer.js#L145-L148

I wasn't aware of the `dispose` function in `atlas-viewer-controls.js` so we should probably make a call to that one too when disposing the main controller. But It looks like they are for different events. Not sure if that answers your question.

Reply:
Thanks; it does seem that for `touch` events the listener might still be on.
Would it make sense to add and remove these controller-related event listeners from the same place?

Reply:
It looks like maybe the original intention was to separate events that would directly interact with the map and others such as window resizing in another. I admit some of these look like they are similar enough but I'm not sure if it makes sense to put them all in the same place right now. I do think there is room for some refactoring work for this repo in general. Overall I tried to introduce just one new function that's exposed by the library for the specific problem and I was hoping that it didn't add any new complexity in terms of future maintenance.

Reply:
Actually, I just remembered that we are setting `cameraControls` to null [here](https://github.com/MetabolicAtlas/3d-network-viewer/blob/8e392604a6f161e1b819e8d7d2718135bea2b855/src/met-atlas-viewer.js#L1092) so I think it might be automatically detaching the event listeners in `atlas-viewer-controls.js` since the listener object itself is removed.

Reply:
Would it make sense to make a note of this for the refactoring?

Reply:
Yes, added a new issue now and linked to this discussion thread. Thanks!